### PR TITLE
Add cups-browsed Information Disclosure module

### PR DESCRIPTION
**Pull request by bcoles**
Short and sweet. Simple scanner.

1. Do: `use modules/auxiliary/scanner/misc/cups_browsed_info_disclosure`
2. Do: `set rhosts [ips]`
3. Do: `run`

### CODE REVIEW

File: modules/auxiliary/scanner/misc/cups_browsed_info_disclosure.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    report_host(host: cli.peerhost)
+    report_service(
+      host: cli.peerhost,
+      proto: 'udp',
+      port: rport,
+      name: 'cups-browsed',
+      info: info
+    )
+  end
```
Comment (adfoster-r7) [inline modules/auxiliary/scanner/misc/cups_browsed_info_disclosure.rb:58]: I think we can throw on a `report_vuln` call here to register it as a vulnerability

Comment (bcoles) [reply modules/auxiliary/scanner/misc/cups_browsed_info_disclosure.rb:65]: > I think we can throw on a `report_vuln` call here to register it as a vulnerability

Done

```
msf6 auxiliary(scanner/misc/cups_browsed_info_disclosure) > vulns 

Vulnerabilities
===============

Timestamp                Host             Name                                 References
---------                ----             ----                                 ----------
2024-08-15 21:30:58 UTC  192.168.200.132  cups-browsed information disclosure  URL-https://github.com/OpenPrinting/cups-browsed/secu
                                                                               rity/advisories/GHSA-rj88-6mr5-rcw8,URL-https://www.e
                                                                               vilsocket.net/2024/09/26/Attacking-UNIX-systems-via-C
                                                                               UPS-Part-I/
```

**Comment by adfoster-r7** [comment]
Module against happy path target :+1:

```
msf6 auxiliary(scanner/misc/cups_browsed_info_disclosure) > 
[*] Using URL: http://10.4.231.99:8080/printers/yxoSgVHfImxVtw
[*] Sending probes to 192.168.123.1->192.168.123.1 (1 hosts)
[+] 10.4.231.99: CUPS/2.4.2 (Linux 6.10.4-linuxkit; x86_64) IPP/2.0
[+] 10.4.231.99: CUPS/2.4.2 (Linux 6.10.4-linuxkit; x86_64) IPP/2.0
[+] 10.4.231.99: CUPS/2.4.2 (Linux 6.10.4-linuxkit; x86_64) IPP/2.0
```

Database checks :+1:

```
msf6 auxiliary(scanner/misc/cups_browsed_info_disclosure) > hosts

Hosts
=====

address      mac  name  os_name  os_flavor  os_sp  purpose  info  comments
-------      ---  ----  -------  ---------  -----  -------  ----  --------
10.4.231.99             Unknown                    device

msf6 auxiliary(scanner/misc/cups_browsed_info_disclosure) > vulns

Vulnerabilities
===============

Timestamp                Host         Name                                 References
---------                ----         ----                                 ----------
2024-09-30 20:42:57 UTC  10.4.231.99  cups-browsed Information Disclosure  URL-https://github.com/OpenPrinting/cups-browsed/security/advisories/GHSA-rj88-6mr5-rcw8,URL-https://www.evilsocket.net/2024/09/26/Attacking-UNIX-systems-
                                                                           via-CUPS-Part-I/

msf6 auxiliary(scanner/misc/cups_browsed_info_disclosure) > services
Services
========

host         port  proto  name          state  info
----         ----  -----  ----          -----  ----
10.4.231.99  633   udp    cups-browsed  open   CUPS/2.4.2 (Linux 6.10.4-linuxkit; x86_64) IPP/2.0
```

(Haven't tested against a non-vuln target, just dumping this output here as I'm in the area)

File: documentation/modules/auxiliary/scanner/misc/cups_browsed_info_disclosure.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+## Vulnerable Application
+
+Retrieve CUPS version and kernel version information from `cups-browsed` services.
+
+
```
[the remark this answers is not in the document]
Comment (bcoles) [reply documentation/modules/auxiliary/scanner/misc/cups_browsed_info_disclosure.md:5]: `cups-browsed` is unlikely to work without a functioning CUPS environment.

I'll add this if you can verify it works. I can't easily test this now.

**Comment by bwatters-r7** [comment]
# Release Notes
Adds scanner module to retrieve CUPS version and kernel version information from `cups-browsed` services.

**Review by bwatters-r7 — APPROVED** [review]

**Pull request merged**